### PR TITLE
add additional checks for compliace operator scans

### PR DESCRIPTION
**Pull request by ycao56**
With this PR we can achieve the goal to further integrate with compliance operator to report remediation. However, we lost the precise event message from scansettingbindings as now we use configuration policy to create it.

1. update compliance operator to watch for openshift-compliance ns
2. use configuration policy to create scansettingbindings
3. add additional policy templates to 
   1. check if the scan is complete
   2. if there is any failed result

image

image

### CODE REVIEW

File: community/CA-Security-Assessment-and-Authorization/policy-compliance-operator-install-upstream.yaml [hunk 1]
```diff
@@ -69,9 +69,8 @@ spec:
                   name: compliance-operator
                   namespace: openshift-compliance
                 spec:
-                  selector:
-                    matchLabels:
-                      policy.open-cluster-management.io/isClusterNamespace: "true"
```
Comment (JAORMX) [inline community/CA-Security-Assessment-and-Authorization/policy-compliance-operator-install-upstream.yaml:74]: Why was this removed? Don't we have this in RHACM nowadays?

File: community/CM-Configuration-Management/policy-compliance-operator-e8-scan.yaml [hunk 2]
```diff
@@ -14,21 +14,67 @@ spec:
   disabled: false
   policy-templates:
     - objectDefinition:
-        apiVersion: compliance.openshift.io/v1alpha1
-        kind: ScanSettingBinding
+        apiVersion: policy.open-cluster-management.io/v1
+        kind: ConfigurationPolicy
```
Comment (JAORMX) [inline community/CM-Configuration-Management/policy-compliance-operator-e8-scan.yaml:18]: The whole integration work was to have the compliance-operator issue events directly to RHACM. This removes that capability and relies on another controller which is not what this policy was meant to do.

**Review by JAORMX — CHANGES_REQUESTED** [review]

File: community/CA-Security-Assessment-and-Authorization/policy-compliance-operator-install-upstream.yaml (hunk 1 above)
Comment (ycao56) [reply community/CA-Security-Assessment-and-Authorization/policy-compliance-operator-install-upstream.yaml:74]: discussed offline that we are going to switch to use configuration policy to create ScanSettingBinding

File: community/CM-Configuration-Management/policy-compliance-operator-e8-scan.yaml (hunk 2 above)
Comment (ycao56) [reply community/CM-Configuration-Management/policy-compliance-operator-e8-scan.yaml:18]: discussed offline that we are going to switch to use configuration policy to create ScanSettingBinding

File: community/CM-Configuration-Management/policy-compliance-operator-e8-scan.yaml [hunk 3]
```diff
@@ -14,21 +14,68 @@ spec:
...
+                metadata:
+                  name: e8
+                  namespace: openshift-compliance
+                status:
+                  phase: DONE
+    - objectDefinition:
+        apiVersion: policy.open-cluster-management.io/v1
+        kind: ConfigurationPolicy
+        metadata:
+          name: compliance-suite-e8-results
```
Comment (ycao56) [inline community/CM-Configuration-Management/policy-compliance-operator-e8-scan.yaml:65]: @JAORMX switched to check for `ComplianceCheckResult` CRs with label `                 compliance.openshift.io/check-status: FAIL`

**Review by gparvin — APPROVED** [review]
Approving changes discussed in call with compliance team.

**Pull request merged**